tls/s2n_cipher_suites.c

```diff
@@ -1155,24 +1155,179 @@ int s2n_set_cipher_as_client(struct s2n_connection *conn, uint8_t wire[S2N_TLS_C
     return 0;
 }
 
-static int s2n_wire_ciphers_contain(const uint8_t *match, const uint8_t *wire, uint32_t count, uint32_t cipher_suite_len)
+static int s2n_wire_ciphers_contain(const uint8_t *match, struct s2n_stuffer *wire_stuffer, uint32_t count, uint32_t cipher_suite_len)
 {
+    uint8_t theirs[S2N_SSLv2_CIPHER_SUITE_LEN] = { 0 };
+    POSIX_ENSURE_LTE(cipher_suite_len, sizeof(theirs));
+
     for (uint32_t i = 0; i < count; i++) {
-        const uint8_t *theirs = wire + (i * cipher_suite_len) + (cipher_suite_len - S2N_TLS_CIPHER_SUITE_LEN);
+        POSIX_GUARD(s2n_stuffer_read_bytes(wire_stuffer, theirs, cipher_suite_len));
 
         if (!memcmp(match, theirs, S2N_TLS_CIPHER_SUITE_LEN)) {
+            POSIX_GUARD(s2n_stuffer_reread(wire_stuffer));
             return 1;
         }
     }
 
+    POSIX_GUARD(s2n_stuffer_reread(wire_stuffer));
     return 0;
 }
 
+static S2N_RESULT s2n_chacha20_is_in_cipher_preferences(const struct s2n_cipher_preferences *cipher_preferences, bool *chacha20_is_present) {
```

Comment:
Should it be possible for cipher preferences to turn on chacha20 boosting without including chacha20? Why do we need to verify this at runtime?

Check out [s2n_security_policies_init](https://github.com/aws/s2n-tls/blob/main/tls/s2n_security_policies.c#L911).

Reply:
It shouldn't be possible but I added it as a sanity check. Is the suggestion here to move this to init?

Reply:
Yes. Unless there's a reason you need to perform the sanity check at runtime?

Reply:
Nope. Moved to init